File: Java/factorial.java
public class factorial {
    public static long factorial(int x){
        long fact=1;
        for(int i=0;i<x;i++){
            fact=fact*i;
        }
        return fact;
    }
    public static void main(String[] args){
        long c=factorial(10);
        System.out.print(c);
    }
}

File: C++/factorial.cpp
#include <iostream>
#include <limits>
using namespace std;

long long factorial(int num) {
    long long fact = 1;
    for (int i = 1; i <= num; ++i)
        fact *= i;
    return fact;
}

// Input validation function
int getValidatedInput(const string &prompt, int minValue) {
    int num;
    cout << prompt;
    while (!(cin >> num) || num < minValue) {
        cin.clear(); // clear error flag
        cin.ignore(numeric_limits<streamsize>::max(), '\n'); // discard invalid input
        cout << "Invalid input! " << prompt;
    }
    return num;
}

int main() {
    int num = getValidatedInput("Enter a non-negative integer: ", 0);
    cout << "Factorial of " << num << " is " << factorial(num) << endl;
    return 0;
}